Frontend/ui-driver/src/Screens/PaymentHistoryScreen/Transformer.purs

```diff
@@ -68,12 +68,12 @@ buildTransactionDetails (API.HistoryEntryDetailsEntityV2Resp resp) gradientConfi
                   {
                     key : "FEE_BREAKUP",
                     title : getString FEE_BREAKUP,
-                    val : if (resp.feeType == AUTOPAY_REGISTRATION || not showFeeBreakup) then "" else (getFeeBreakup offerAndPlanDetails driverFee'.totalRides <> " "<> getString GST_INCLUDE)
+                    val : if (resp.feeType == AUTOPAY_REGISTRATION || not showFeeBreakup) then "" else (getFeeBreakup driverFee'.maxRidesEligibleForCharge (driverFee'.planAmount - (fromMaybe 0.0 driverFee'.totalSpecialZoneCharges)) driverFee'.totalRides)
```

Comment:
Here we are subtracting specialZoneCharges from totalPlanAmount. If totalSpecialZoneCharges is nothing then we are assuming that it's zero.